app/assets/javascripts/autocomplete.coffee

```diff
@@ -0,0 +1,69 @@
+delay = do ->
+  timer = 0
+  (callback, ms) ->
+    clearTimeout timer
+    timer = setTimeout(callback, ms)
+    return
+
+update_textfield = (list_element) ->
+  # remove focused state
+  $(list_element).removeClass('suggestion-focused')
+  $('#autocomplete-textfield').val($(list_element).find('.res-name')[0].innerHTML)
+
+ready = ->
+  # TODO: test with IE 8
+  # onpropertychange -> oninput for IE 8
+  # https://msdn.microsoft.com/en-us/library/ms536956(v=vs.85).aspx
+  $('#autocomplete-textfield').on('focus input propertychange', ->
```

Comment:
I'm also not a huge fan of `id` based javascript bindings.

Generally what I do is add another class to the DOM object and add a `js-` prefix to it.

Example.

```html
<dd><input class="textfield js-autocomplete-textfield" ...>...</dd>
```

So then your coffee would look like
```coffee
$('.js-autocomplete-textfield').on('focus input propertychange', ->
```